apicurio-registry-java-sdk published classes require JDK17

<!--

In order for us to better investigate the bug, please fill in as much relevant information as is reasonable in the format below.

Before proceeding, consider the following:

- This repository is for issues with Apicurio Registry. Make sure the bug report is not more appropriate or another repository.
- Make sure the bug is in Apicurio Registry and not in an external library or other non-Apicurio software.
 For example, we provide a compatibility REST API for Confluent Schema Registry clients. Make sure the bug is not in one of their libraries.
- Run the Registry with DEBUG logging enabled, by passing the LOG_LEVEL=DEBUG env. variable.

-->

## Description

**Registry
Version**: 3.0.0.M3
**Persistence type**: postgresql

We have a lot of users of Flink, and [the current stable version of Flink considers JDK17 experimental](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/java_compatibility/) which means most of our users run this on JDK11.

While attempting to build a client wrapper library for these users, I notice that the artifacts published to maven central for 3.0.0.M3 only work with JDK17+

While this is the first such error we see (from the Kiota generated classes in `apicurio-registry-java-sdk`), our wrapper library will also depend on `apicurio-registry-schema-resolver` and `apicurio-registry-serdes-*` so at least those too should be compatible with JDK11.

Our build pipeline cannot invoke Kiota, so we rely on the maven central published artifacts for these libraries.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project client: Compilation failure: Compilation failure:
[ERROR] /<REDACTED>/schemaregistry/FlinkRegistryClient.java:[3,40] cannot access io.apicurio.registry.rest.client.RegistryClient
[ERROR]   bad class file: /<REDACTED>/maven-m2/io/apicurio/apicurio-registry-java-sdk/3.0.0.M3/apicurio-registry-java-sdk-3.0.0.M3.jar(/io/apicurio/registry/rest/client/RegistryClient.class)
[ERROR]     class file has wrong version 61.0, should be 55.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
[ERROR] /<REDACTED>/schemaregistry/FlinkRegistryClient.java:[28,19] cannot find symbol
```

please publish artifacts that work with JDK11 since a large number of potential users are still using that runtime.

### Environment

any environment with openjdk11 runtime

--- a/java-sdk/pom.xml
+++ b/java-sdk/pom.xml
@@ -27,6 +27,8 @@
     <projectRoot>${project.basedir}/..</projectRoot>
     <kiota.libs.version>1.3.0</kiota.libs.version>
     <kiota.community.version>0.0.18</kiota.community.version>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <kiota.base.url>https://github.com/microsoft/kiota/releases/download</kiota.base.url>
   </properties>
   <dependencies>

